Dockerfile

```diff
@@ -2,9 +2,12 @@ FROM golang:1.14.1-alpine3.11 as builder
 
 RUN apk add --update --no-cache ca-certificates tzdata git make bash && update-ca-certificates
 
-ADD . /opt
 WORKDIR /opt
 
+COPY go.* Makefile /opt/
+RUN make vendor
```

Comment:
This project already vendors it's dependencies, why do we `make vendor` again before building? If we don't trust the deps we committed then we should do a `make vendor && git diff --exit-code` as part of the CI. Otherwise, we should definitely use the checked-in vendor code for our builds. No?

Reply:
Yes, we already implicitly call `vendor` before we build. That's why I want to this explicitly. Also, we pass the vendor dir along so this is just like an integrity check. This additional thing won't hurt, will it? It definitely speeds things up.

Reply:
mmm i'd say let's keep this PR focused; this is about ` Only trigger release job for SemVer tags` so let's put this unrelated change into a new PR.

Also, why do we even do another `make vendor` before building if we know that the deps are checked into the repo? CI does an integrity check with `make vendor && git diff --exit-code`. There should be no reason for another explicit  `make vendor`, instead, we should declare the vendor dir and go.* files deps of the binary build target, so that `go mod tidy && go mod vendor` is only run when actually necessary. WDYT? Otherwise what we should really do IMO is not commit our vendor dir, which is also a valid path

Reply:
This is what we already have. Then I'm reverting the changes and we're good.